include/fields/DateTimeField.php

```diff
@@ -426,13 +426,14 @@ private static function sanitizeDate($value, $user)
 		}
 
 		$value = str_replace('T', ' ', $value);
-		list($date, $time) = explode(' ', $value);
+		list($date, $time) = array_pad(explode(' ', $value), 2, '');
 		if (!empty($date) && !in_array($time, ['AM', 'PM'])) {
-			$date = self::__convertToDBFormat($date, $user->date_format);
-			$value = $date;
-			if (!empty($time)) {
-				$value .= ' ' . rtrim($time);
+			if (empty($time)) {
+				$date = self::__convertToDBFormat($date, $user->date_format);
+			} else {
+				$date = self::__convertToDBFormat($date.' ' . rtrim($time), $user->date_format);
```

Comment:
i know. Current results are not affected. It ads the possibility of having time information also. 

In a PR to come i will send in my modificated date(time) converter function with much less code than before and also be able to handle time - if provided.

This file (and its also called) by "datetime" fields (not only date fields). However, the time information is truncated. Till now nobody seems to recognized this, because later on it will automaticly set current date and time. I got on this because of sending datetime information (e.g. 'createdtime') via API and all it ends on was using current date and time.